doc/Data Plane Code Generation/Subclients.md

```diff
@@ -0,0 +1,233 @@
+# Customizing the generated code
+
+## Subclients
+
+There are two categories of clients: service clients and their subclients. Service clients can be instantiated and have the Client suffix. Subclients can only be created by calling factory methods on other clients (commonly on service clients) and do not have the client suffix. The service client is the entry point to the API for an Azure service. Please refer to [guideline](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-subclients) for more detail information.
+
+There are two tools to organize subclients and service client.
+
+- **Single top-level client**: define one top-level service client with the name <service_name>Client, all other clients will be its subclient.
+
+- **Client hierarchy**: organize the client service and subclients via defining the parent-subclient hierarchy dependency.
+
+**Note**: You can combine those two approaches to define an client hierarchy with one top-level client and its subclient may have sub clients also. 
+
+### Single top-level client
+
+With multiple service clients it may be better to group them all under one top-level client.
+
+<details>
+
+**Generated code before:**
+
+``` C#
+//Generated\Client1Client.cs
+namespace Azure.Service.SubClients
```

Comment:
Can we use a real service like Purview for this case?  I feel a bit difficult to understand the concept here like
the name space is call Azure.Service.SubClients, but the Client1Client should be a service client. And that name is also a bit strange

Reply:
I would love this!  I would love if we could start thinking in terms of the impact generator changes have to service libraries more than just the autorest.csharp test output.  The test client output helps us explore edge cases and get a preview of what updates to the generator will look like in the azure-sdk-for-net repo, but they are hard to understand outside of the autorest context, and we shouldn't expect service team partners to try to learn these.

Reply:
Hi @annelo-msft  @lirenhe  we are familiar with some real service like Purview, in that case, it may be more easy to understand if we take Puerview as example. But actually the reader of the document will have no ideal about Purview, and the complex of the Purview service may make the document more difficult to understand. The document is to show subclient feature, I think it is better to use a sample RP with simple client structure, then it is more easy to show the subclient principle.  By the way, customer need to understand what is subclient then he can understand the document, so in the beginning we add subclient guideline reference link.

Reply:
I don't think user need to understand Purview to understand the subclient. For me, the name like 'Client1Client' make me feel confused.  The PR https://github.com/Azure/azure-sdk-for-net/pull/27558  also use a MetricAdvisor as an example and I didn't see any difficultiy to understand it.

Reply:
update to use a real service Purview Account for example.